src/message/ReadReceiptsScreen.js

```diff
@@ -0,0 +1,170 @@
+/* @flow strict-local */
+
+import React, { useCallback, useContext, useMemo } from 'react';
+import type { Node } from 'react';
+import { View, FlatList } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import useFetchedDataWithRefresh from '../common/useFetchedDataWithRefresh';
+import ZulipTextIntl from '../common/ZulipTextIntl';
+import { useGlobalSelector, useSelector } from '../react-redux';
+import * as api from '../api';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import type { RouteProp } from '../react-navigation';
+import Screen from '../common/Screen';
+import { useConditionalEffect, useHasNotChangedForMs, useHasStayedTrueForMs } from '../reactUtils';
+import { getAuth, getZulipFeatureLevel } from '../account/accountsSelectors';
+import { showToast } from '../utils/info';
+import { TranslationContext } from '../boot/TranslationProvider';
+import UserItem from '../users/UserItem';
+import { tryGetUserForId } from '../users/userSelectors';
+import type { UserId } from '../api/idTypes';
+import { getGlobalSettings } from '../directSelectors';
+import type { UserOrBot } from '../api/modelTypes';
+import LoadingIndicator from '../common/LoadingIndicator';
+import WebLink from '../common/WebLink';
+import { createStyleSheet } from '../styles';
+import ZulipText from '../common/ZulipText';
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'read-receipts'>,
+  route: RouteProp<'read-receipts', {| +messageId: number |}>,
+|}>;
+
+export default function ReadReceiptsScreen(props: Props): Node {
+  const { navigation } = props;
+  const { messageId } = props.route.params;
+
+  const auth = useSelector(getAuth);
+  const zulipFeatureLevel = useSelector(getZulipFeatureLevel);
+  const language = useGlobalSelector(state => getGlobalSettings(state).language);
+  const _ = useContext(TranslationContext);
+
+  const callApiMethod = useCallback(
+    () => api.getReadReceipts(auth, { message_id: messageId }, zulipFeatureLevel),
+    [auth, messageId, zulipFeatureLevel],
+  );
+
+  const { latestResult, latestSuccessResult } = useFetchedDataWithRefresh(callApiMethod, 15_000);
+
+  // TODO: These vanishing toasts are easy to miss. Instead, show
+  //   latestResultIsError, isFirstLoadLate, haveStaleData, and onPressUser
```

Comment:
Is `onPressUser` in this list unintentionally? Or, what would it mean to "show [it] with something more persistent"?

Reply:
Ah, haha—yeah, that was accidental. I grabbed names from four code stanzas just below this, when I should've grabbed from just the first three.